src/picknik_ur_gazebo_config/launch/sim/hardware_sim.launch.py

```diff
@@ -81,31 +103,57 @@ def generate_simulation_description(context, *args, **settings):
 
     # Launch Gazebo.
     print(f"Starting Gazebo with world {world_name}")
-    print(f"GUI: {use_gui}, Verbose: {is_verbose}")
+    print(f"GUI: {use_gui}, Verbose: {is_verbose}, Test mode: {is_test}")
 
     sim_args = f"-r --render-engine {gz_renderer}"
     if is_verbose:
         sim_args += " -v 4"
     if not use_gui:
         sim_args += " -s --headless-rendering"
 
+        # If no display is available, set up a Xvfb for headless rendering
+        if not os.environ.get("DISPLAY"):
+            print("Adding Xvfb for simulation...")
+            display_id = ":99"
+            os.environ["DISPLAY"] = display_id
+            xvfb = ExecuteProcess(
+                name="xvfb", cmd=["Xvfb", display_id, "-screen", "0", "1600x1200x16"]
+            )
+            nodes.append(xvfb)
+
     gazebo = IncludeLaunchDescription(
         get_launch_file("ros_gz_sim", "launch/gz_sim.launch.py"),
         launch_arguments=[("gz_args", [f"{sim_args} {modified_world_file}"])],
     )
-    return [gazebo]
+    nodes.append(gazebo)
+    return nodes
 
 
 def generate_launch_description():
     system_config_parser = SystemConfigParser()
-    optional_feature_setting = system_config_parser.get_optional_feature_configs()
     cameras_config = system_config_parser.get_cameras_config()
+    optional_feature_setting = system_config_parser.get_optional_feature_configs()
+
+    # Launch arguments
+    is_test_arg = DeclareLaunchArgument(
+        name="is_test",
+        default_value="false",
+        description="If true, declares that the launch should be configured for testing.",
+    )
 
     # The path to the auto_created urdf files
     robot_urdf = system_config_parser.get_processed_urdf()
+
     robot_urdf_ignition = path_pattern_change_for_ignition(robot_urdf)
 
-    # Launch Gazebo
+    # Include URDF
+    scene_xacro_path = get_ros_path(
+        "picknik_ur_gazebo_config", "description/simulation_scene.urdf.xacro"
+    )
+    scene_urdf = xacro_to_urdf(scene_xacro_path, None)
+    scene_urdf_ignition = path_pattern_change_for_ignition(scene_urdf)
```

Comment:
this code is duplicated in that I had moved it below, near where it's use for launching. You can undo this part, or remove the other below. Either/or